### PR TITLE
Initial Dishwasher Support

### PR DESCRIPTION
**Pull request by simbaja**
This pull request adds initial dishwasher support to the gekitchen library.  There are a couple anomalies that I saw (like an unexpected value for the rinse agent), but it seems to work for my dish washer.

### CODE REVIEW

File: gekitchen/erd/types/__init__.py [hunk 1]
```diff
@@ -0,0 +1,4 @@
+"""GE ERD Types"""
+
+from .fridge import *
+from .oven import *
```
Comment (ajmarks) [inline gekitchen/erd/types/__init__.py]:
```suggestion
from .oven import *

```

File: gekitchen/erd/values/__init__.py [hunk 2]
```diff
@@ -0,0 +1,6 @@
+"""GE ERD Values"""
+
+from .common import *
+from .dishwasher import *
+from .fridge import *
+from .oven import *
```
Comment (ajmarks) [inline gekitchen/erd/values/__init__.py]:
```suggestion
from .oven import *

```

File: gekitchen/erd_utils.py [hunk 3]
```diff
@@ -373,6 +384,32 @@ def _decode_filter_status(value: str) -> ErdFilterStatus:
...
+
+def _decode_cycle_state(value: str) -> ErdCycleState:
+    """ Decodes the dishwasher cycle state """    
+    try:
+        raw = ErdCycleStateRaw(value)
+        return CYCLE_STATE_RAW_MAP[raw]
+    except ValueError:
+        return ErdCycleState.NA
+    except KeyError:
+        return ErdCycleState.NA
```
Comment (ajmarks) [inline gekitchen/erd_utils.py]:
```suggestion
    except (KeyError, ValueError):
        return ErdCycleState.NA
```

File: gekitchen/erd_utils.py [hunk 4]
```diff
@@ -373,6 +384,32 @@ def _decode_filter_status(value: str) -> ErdFilterStatus:
...
+
+def _decode_rinse_agent(value: str) -> ErdRinseAgent:
+    """ Decodes the dishwasher rinse agent status """
+    try:
+        raw = ErdRinseAgentRaw(value)
+        return RINSE_AGENT_RAW_MAP[raw]
+    except ValueError:
+        return ErdRinseAgent.NA
+    except KeyError:
+        return ErdRinseAgent.NA
```
Comment (ajmarks) [inline gekitchen/erd_utils.py]:
```suggestion
    except (KeyError, ValueError):
        return ErdRinseAgent.NA
```

File: gekitchen/erd_utils.py [hunk 5]
```diff
@@ -77,6 +76,18 @@ def _decode_erd_string(value: str) -> str:
 
     return raw_bytes[1:].decode('ascii')
 
+def _decode_erd_string(value: str) -> str:
+    """
+    Decode an string value sent as a hex encoded string.
+
+    TODO: At least for the dishwasher cycle the first character is not a check sum
+    are there potentially different decoding methods needed?
```
Comment (ajmarks) [inline gekitchen/erd_utils.py]: Looks like you're correct.  I went back to the decompiled java courtesy of [bytecode-viewer](https://github.com/Konloch/bytecode-viewer) on the v1.0.3.13 APK, and found this:
image

Looks like we can kill that TODO, and I'll open an issue for myself on the other.

**Review by ajmarks — CHANGES_REQUESTED** [review]
This looks great.  I haven't checked that the various ERD codes and states are correct, but will try to give it glance over against the decompiled java later.  I think the only real change is to kill that TODO, since it is now done.

File: gekitchen/erd_utils.py (hunk 5 above)
Comment (simbaja) [reply gekitchen/erd_utils.py]: Sounds good - I've got another (bigger) update coming, we can resolve in that one

**Comment by simbaja** [comment]
I had more time today and took a stab at re-organizing the ERD section, breaking it up and adding converters.  It's a bit of a bigger update than I anticipated, let me know if I didn't do something the Python way (or if there's a better way).

**Comment by simbaja** [comment]
Going to re-do the request by making a separate branch so that new commits aren't automatically added...

**Pull request closed without merging**